### PR TITLE
Fix simple docstring formatter idempotence

### PR DESCRIPTION
**Pull request by glenn-jocher**
## Summary
- keep sectionless one-line docstrings on one line in the custom formatter
- add a regression test for the Ruff D200 boundary that caused format/docs automation churn

## Validation
- ruff check --fix --unsafe-fixes --extend-select F,I,D,UP,RUF,FA --target-version py39 --ignore D100,D104,D203,D205,D212,D213,D401,D406,D407,D413,RUF001,RUF002,RUF012 actions/format_python_docstrings.py tests/test_cli_commands.py
- ruff format --line-length 120 actions/format_python_docstrings.py tests/test_cli_commands.py
- python -m pytest tests/test_cli_commands.py
- reproduced against ultralytics PR 22870 tests/test_engine.py: formatter left the file unchanged and ruff check --select D200 passed

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
🛠️ This PR updates the Python docstring formatter to preserve simple docstrings as single-line strings more reliably, and adds a test to ensure that behavior stays consistent.

### 📊 Key Changes
- Simplified the `single_ok` logic in `actions/format_python_docstrings.py`.
- Removed the line-length check from the decision about whether a docstring can remain single-line.
- Kept the existing safeguards that prevent single-line formatting for docstrings containing:
  - multiple lines
  - named sections like `Args:` or similar
  - list-style content
- Added a new test in `tests/test_cli_commands.py` to verify that a simple docstring remains on one line.
- Imported `format_python_docstrings` in the CLI command tests so its formatter behavior is explicitly covered. ✅

### 🎯 Purpose & Impact
- Helps the formatter better align with Ruff’s `D200` style expectations for one-line docstrings. 📏
- Reduces unnecessary reformatting of simple docstrings, making automated formatting more predictable. 🔄
- Improves developer experience by avoiding formatting changes that may conflict with linting or code style tools. 👨‍💻
- Adds test coverage to prevent regressions, making future updates safer and more reliable. 🧪

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @glenn-jocher, thank you for submitting a `ultralytics/actions` 🚀 PR! This is an automated message to help streamline review, and an engineer will assist shortly. Please review the checklist below to help ensure a smooth merge 😊

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/actions/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/actions` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: actions/format_python_docstrings.py [hunk 1]
```diff
@@ -377,12 +377,7 @@ def format_docstring(
     text = content.strip()
     has_section = any(f"{s}:" in text for s in SECTIONS)
     has_list = any(is_list_item(line) for line in text.splitlines())
-    single_ok = (
-        ("\n" not in text)
-        and not has_section
-        and not has_list
-        and (indent + len(prefix) + len(quotes) * 2 + len(text) <= width)
-    )
+    single_ok = ("\n" not in text) and not has_section and not has_list
```
Comment (UltralyticsAssistant) [inline actions/format_python_docstrings.py:380]: ⚠️ **HIGH**: Removing the width check changes more than idempotence: every simple docstring now stays single-line even when it exceeds the configured `width`. That means the formatter can emit arbitrarily long docstring lines and stop wrapping cases it previously handled. If the bug was in the fit calculation, this needs a narrower fix than dropping the guard entirely.

File: tests/test_cli_commands.py [hunk 2]
```diff
@@ -16,3 +17,10 @@ def test_importable_modules():
     assert hasattr(summarize_pr, "main")
     assert hasattr(summarize_release, "main")
     assert hasattr(update_markdown_code_blocks, "main")
+
+
+def test_docstring_formatter_keeps_simple_docstrings_single_line():
```
Comment (UltralyticsAssistant) [inline tests/test_cli_commands.py:22]: 💡 **MEDIUM**: This regression test only covers a docstring that already fits within the 120-column limit, so it won't catch the behavioral change introduced above for over-width simple docstrings. Please add a companion case asserting that a simple docstring longer than `width` still wraps, otherwise this test suite will miss the width-regression entirely.

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

The idempotence fix is heading in the right direction, but the current implementation removes the formatter's line-width safeguard for all simple docstrings. That looks like a real behavior regression, and the added test does not cover it.

💬 Posted 2 inline comments

**Comment by UltralyticsAssistant** [comment]
Merged, nice work @glenn-jocher! 🚀

As Leonardo da Vinci said, *“Simplicity is the ultimate sophistication.”* This update is a great example of that in practice: cleaner single-line docstring handling, better alignment with Ruff expectations, and stronger test coverage to keep formatter behavior predictable over time.

Thanks for continuing to refine the developer experience with thoughtful, high-impact improvements.
